tests/drivers/can/utilities/src/main.c

```diff
@@ -135,13 +149,58 @@ static void test_zcan_filter_to_can_filter(void)
 	zassert_equal(filter.can_mask, expected.can_mask, "CAN mask not same");
 }
 
+/**
+ * @brief Test of @a can_dlc_to_bytes()
+ */
+static void test_can_dlc_to_bytes(void)
+{
+	uint8_t dlc;
+
+	/* CAN 2.0B/CAN-FD DLC, 0 to 8 data bytes */
+	for (dlc = 0; dlc <= 8; dlc++) {
+		zassert_equal(can_dlc_to_bytes(dlc), dlc, "wrong number of bytes for DLC %u", dlc);
+	}
+
+	/* CAN-FD DLC, 12 to 64 data bytes in steps */
+	zassert_equal(can_dlc_to_bytes(9),  12, "wrong number of bytes for DLC 9");
+	zassert_equal(can_dlc_to_bytes(10), 16, "wrong number of bytes for DLC 10");
+	zassert_equal(can_dlc_to_bytes(11), 20, "wrong number of bytes for DLC 11");
+	zassert_equal(can_dlc_to_bytes(12), 24, "wrong number of bytes for DLC 12");
+	zassert_equal(can_dlc_to_bytes(13), 32, "wrong number of bytes for DLC 13");
+	zassert_equal(can_dlc_to_bytes(14), 48, "wrong number of bytes for DLC 14");
+	zassert_equal(can_dlc_to_bytes(15), 64, "wrong number of bytes for DLC 15");
+}
```

Comment:
You could consider additionally adding one test for an out of range dlc like 16 or higher.